### PR TITLE
[redesign] Add Identity management

### LINKED ISSUE #1340: [redesign] User account: identity tab
Implement the identity tab for the user account as per design specs

### PR DESCRIPTION
**Pull request by tiagoalvesdulce**
Closes #1340

### CODE REVIEW

**Comment by tiagoalvesdulce** [comment]
1. WIll fix, opened de PR to get help with the verificationtoken bit.
2. OK
3. No need to worry about it, was just testing solutions for the verificationtoken problem.
4. Will not fix in this PR, will open another one after this. Will focus on Identity Mngmt.
5. Couldn't find it in the specs.
6. IMO we should hide it with 1, people can see the user id in the URL anyways.
7. It was confusing in the specs, I understood we should show some os the keys and them show all of them in the Modal, but I'm OK with removing it.
8. OK

File: src/pages/User/VerifyKey.jsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  } = useVerifyKey();
+
+  useEffect(() => {
+    const { verificationtoken } = qs.parse(location.search);
+    if (
+      isEmpty(location.search) ||
+      !verificationtoken ||
+      typeof verificationtoken !== "string"
+    ) {
+      history.push("/user/login");
```
Comment (fernandoabolafio) [inline src/pages/User/VerifyKey.jsx]: Better throw an error here instead of redirecting to login without further explanation.

Comment (tiagoalvesdulce) [reply src/pages/User/VerifyKey.jsx]: Yeah, I agree.

**Comment by tiagoalvesdulce** [comment]
@MariaPleshkova can you help me with the fifth item of Fernando's list? Missing text copy for Past Public Keys and User ID. Thanks!

**Comment by MariaPleshkova** [comment]
@tiagoalvesdulce yeah, there's no text copy in the specs. I think we need to request it

**Comment by tiagoalvesdulce** [comment]
OK, everything but the copies is done.

File: src/pages/User/VerifyKey.jsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  useEffect(() => {
+    if (loggedInAsEmail && !keyUpdated) {
+      const { verificationtoken } = qs.parse(location.search);
+      onVerifyUserKey(loggedInAsEmail, verificationtoken);
+    }
+  }, [loggedInAsEmail, location, onVerifyUserKey, keyUpdated]);
+
+  useEffect(() => {
+    if (
+      verifyUserKey &&
```
Comment (fernandoabolafio) [inline src/pages/User/VerifyKey.jsx]: Can we extract this big if statement to a more meaningful constant? Also, it is very confusing that we have two effects in this component and one is calling `onVerifyUserKey` and the other one is calling  `verifyUserKey`. Perhaps we can at least give better names to the functions to become more clear what they are doing. 
If I am not missing anything here you could also simplify whats happening here by calling the `onVerifyUserKey` using an async function and if it succeeds you can immediately verify the new key and update the local storage.
Something like:

```jsx
const VerifyKey = ({}) => {
   ...
   const verifyAndUpdateLocalKey = useCallback(async () => {
       setLoading(true);
       try {
         await onVerify(...)
         verifyUserPubkey(...)
         pki.myPubKeyHex(...)
         setLoading(false)
       } catch(e) {
          setLoading(false)
          throw e;
       }
    },[...])

    useEffect(() => {
       if(...) {
        verifyAndUpdateLocalKey(...)
      }
    },[...])


   return (...);
}
```

This removes the need for two effects on this small component and makes the code more readable. Also, notice that I am adding a flag for loading which is currently lacking.

File: src/componentsv2/ModalImportIdentity.jsx [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+      onSubmitFiles(onIdentityImported, userPubkey, loggedInAsEmail, values, setFileError);
+      resetForm();
+      setSubmitting(false);
+    } catch (e) {
+      setSubmitting(false);
+      setFieldError("global", e);
+    }
+  };
+  return (
+    <Modal title="Import Identity" show={show} onClose={onClose}>
```
Comment (fernandoabolafio) [inline src/componentsv2/ModalImportIdentity.jsx]: Please, set a fixed width for the Modal so it doesn't expand when an error is presented.

File: src/pages/User/VerifyKey.jsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    verifyUserKey,
+    updateMe,
+    verifyUserKeyError,
+    onVerifyUserKey
+  } = useVerifyKey();
+
+  const [keyUpdated, setKeyUpdated] = useState(false);
+
+  useEffect(() => {
+    if (loggedInAsEmail && !keyUpdated) {
```
[the remark this answers is not in the document]
Comment (tiagoalvesdulce) [reply src/pages/User/VerifyKey.jsx]: Really? They are not related. `keyUpdated` is just a check to not fetch an already updated key and the loggedInAsEmail check is to not pass an `undefined` value to `onVerifyUserKey`. I can not think of a good name for this constant, to be honest. Although, I'm free to listen to suggestions if you have one.

File: src/componentsv2/ModalImportIdentity.jsx (hunk 3 above)
Comment (tiagoalvesdulce) [reply src/componentsv2/ModalImportIdentity.jsx]: Are you sure? Setting a fixed width for the Modal doesn't sound very responsive.

Comment (tiagoalvesdulce) [reply src/componentsv2/ModalImportIdentity.jsx]: I tested and responsiveness works just fine

File: src/pages/User/VerifyKey.jsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (fernandoabolafio) [reply src/pages/User/VerifyKey.jsx]: Ok, let's keep it like that.

**Comment by MariaPleshkova** [comment]
@fernandoabolafio @tiagoalvesdulce added this as a new issue #1377

**Pull request merged**